gym/vector/vector_env.py

```diff
@@ -0,0 +1,70 @@
+import gym
+from gym.spaces import Tuple
+from gym.vector.utils.spaces import batch_space
+
+__all__ = ['VectorEnv']
+
+
+class VectorEnv(gym.Env):
+    """Base class for vectorized environments.
+
+    Parameters
+    ----------
+    num_envs : int
+        Number of environments in the vectorized environment.
+
+    observation_space : `gym.spaces.Space` instance
+        Observation space of a single environment.
+
+    action_space : `gym.spaces.Space` instance
+        Action space of a single environment.
+    """
+    def __init__(self, num_envs, observation_space, action_space):
+        super(VectorEnv, self).__init__()
+        self.num_envs = num_envs
+        self.observation_space = batch_space(observation_space, n=num_envs)
+        self.action_space = Tuple((action_space,) * num_envs)
+
+        self.closed = False
+        self.viewer = None
+
+        # The observation and action spaces of a single environment are
+        # kepts in separate properties
+        self.single_observation_space = observation_space
+        self.single_action_space = action_space
+
+    def reset_async(self):
+        pass
+
+    def reset_wait(self, **kwargs):
+        raise NotImplementedError()
+
+    def reset(self):
+        self.reset_async()
+        return self.reset_wait()
+
+    def step_async(self, actions):
+        pass
+
+    def step_wait(self, **kwargs):
+        raise NotImplementedError()
+
+    def step(self, actions):
+        self.step_async(actions)
+        return self.step_wait()
+
+    def close_extras(self):
```

Comment:
While close_extras is in the baselines API, I'm not certain it's much better than putting:

```
if self.closed:
    return
<close stuff>
super().close()
```

In subclasses.  Is close_extras actually that much better?